src/graphql/resolvers/custom/patient.js

```diff
@@ -71,24 +88,38 @@ module.exports = {
                     info,
                     'Patient'
                 );
-                if (patients.length === 0) {
+                if (patients && patients.length === 0) {
                     throw new Error(`Patient not found ${args.patientId}`);
                 }
                 const patientToChange = patients[0];
-                const practitioners = await getResources(
-                    parent,
-                    {
-                        ...args,
-                        id: args.practitionerId,
-                    },
-                    context,
-                    info,
-                    'Practitioner'
-                );
-                if (practitioners.length === 0) {
-                    throw new Error(`Practitioner not found ${args.practitionerId}`);
+                if (deletePractitioner && patientToChange.generalPractitioner === null){
```

Comment:
you're explicitly checking against null so you won't catch the undefined scenario.  is that intentional? https://levelup.gitconnected.com/javascript-null-vs-undefined-2acda986f79f

Reply:
good catch i missed that i will fix